br/pkg/task/restore.go

```diff
@@ -769,6 +753,28 @@ func runRestore(c context.Context, g glue.Glue, cmdName string, cfg *RestoreConf
 		}()
 	}
 
+	if isFullRestore(cmdName) {
+		// we need check cluster is fresh every time. except restore from a checkpoint.
+		if client.IsFull() && len(checkpointSetWithTableID) == 0 {
+			if err = client.CheckTargetClusterFresh(ctx); err != nil {
```

Comment:
This will make BR fail once there are any table in the cluster. (Even there isn't intersection of them and the backup archive.) Will this break some use cases?

Reply:
Yes, It's intended  @benmaoer has confirmed it. so this behaviour will introduced in v7.3. and won't cherry-pick back to any release versions.

Reply:
Yes,  there are 2 options:

1.  When restoring the entire cluster from a full backup, it is necessary to check whether the target cluster is an empty cluster, which has no user tables. 

2. When restoring specific tables or databases from the backup, it is necessary to detect if there are any conflicts between the tables to be restored and the existing tables in the target cluster. If conflicts are found, an error should be reported in advance to avoid data integrity issues.

Reply:
Sometimes `br restore full` don't mean I realy want to resotre the entire cluster. In certain scenarios, I just want to restore multiple repositories, but they are distributed across different databases, or just want to use regular expressions.
So, can you add a `--force` option to skip this check??

Reply:
I think maybe we should skip the check if user has specified `--filter` argument already. @benmaoer @YuJuncen  WDYT?